File: app/src/main/java/com/lg/dome/mvvm/MVVMModel.java
package com.lg.dome.mvvm;

import androidx.databinding.ObservableField;

import java.util.Random;

/**
 * Created by lqj on 2020/7/20.
 */
public class MVVMModel {
    public ObservableField<Account> account = new ObservableField<>();
    //模拟查询账号数据
    public void getAccountData(String accountName, MCallback callback){
        Random random = new Random();
       // boolean isSuccess = random.nextBoolean();
      //  if(isSuccess){
            Account account = new Account();
            account.setName(accountName);
            account.setLevel(random.nextInt());
            callback.onSuccess(account);
      //  }else {
       //     callback.onFailed();
       // }
    }
}

File: app/src/main/java/com/lg/dome/mvvm/MVVMModel.kt
package com.lg.dome.mvvm

import androidx.databinding.ObservableField
import java.util.*

/**
 * Created by lqj on 2020/7/20.
 */
class MVVMModel {
    var account =
        ObservableField<Account>()

    //模拟查询账号数据
    fun getAccountData(accountName: String?, callback: MCallback) {
        val random = Random()
        // boolean isSuccess = random.nextBoolean();
//  if(isSuccess){
        val account = Account()
        account.name = accountName
        account.level = random.nextInt()
        callback.onSuccess(account)
        //  }else {
//     callback.onFailed();
// }
    }
}